sdk/core/azure-core-amqp/docs/architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture Docs
+
+## Qpid Proton-J Integration
+
+Qpid Proton-J publishes events and messages via its event-driven process called [`Reactor`][Reactor]. `azure-core-amqp`
+hooks into Qpid Proton-J Reactor via `Handlers`.  [`BaseHandler`][BaseHandler] contains all of the events that can be
+listened to.  `Handlers` can be associated with classes implementing the `Extendable` interface via
+`BaseHandler.setHandler(Extendable, Handler)`.  Proton-J `Connection`, `Session`, `Link`, `Sender` and `Receiver` are
+all `Extendable`.
+
+The UML diagram below shows this relationship; the interfaces shown in green are Proton-J classes. Each Proton-J
+instance (ie. `Connection`, `Session`, `Sender`, `Receiver`) is associated with one corresponding `*Handler`.  Each
+azure-core-amqp `AmqpConnection` is associated with one [`Reactor`][Reactor].  When that instance closes, the AMQP
+connection is also closed.
+
+Each [ReactorConnection][ReactorConnection] has one Proton-J [`Reactor`][Reactor] instance.  Each [`Reactor`][Reactor]
+has one [`ReactorDispatcher`][ReactorDispatcher] and one [`ReactorExecutor`][ReactorExecutor].
+
+![azure-core-amqp integration with Proton-J][AzureCoreAmpqArchitecture]
+
+## Prefetch and AMQP Link Credits
```

Comment:
This looks great.  One thing to call out is that It would be nice if terms like link credits were defined before their use, similar to what's in : https://github.com/Azure/amqpnetlite/blob/master/docs/articles/building_application.md